### PR TITLE
Migration to `clap 3.0` and additional unit test coverage

### PR DESCRIPTION
**Pull request by kiavash-at-work**
* There are some `Clap 2.0` functions/features that are now deprecated in `Clap 3.0`.
* added Cargo.lock #18.
* changing version to 0.5.x

### CODE REVIEW

**Comment by kiavash-at-work** [comment]
@ashuio , I have started porting the code to `clap 3.0` and fixing some TODO. At this stage, it compiles and passes all the unit tests, but I have not done integration tests (we need to automate that too), so for now I am keeping this PR open.

**Comment by ashuio** [comment]
Thanks for taking the initiative, I'll see how i can help too

File: shavee-core/src/zfs.rs [hunk 1]
```diff
@@ -1,135 +1,97 @@
...
-
-                let mut e: Vec<u8> = Vec::new();
-                zfs.stderr.unwrap().read_to_end(&mut e).unwrap();
-                let error = String::from_utf8_lossy(&e);
-                return Err(error.into());
-            }
-        }
-        Err(error) => return Err(error.into()),
-    };
+#[derive(Debug, PartialEq)]
```
Comment (kiavash-at-work) [inline shavee-core/src/zfs.rs:4]: Converted the zfs module to OO programming to use Struct and Module.

File: shavee-core/src/zfs.rs [hunk 2]
```diff
@@ -1,135 +1,97 @@
...
+pub struct Dataset {
+    pub dataset: String,
 }
+//TODO: Modify the trait function to return _Self_ to enable chain actoions: dataset.loadkey(passphrase).mount()
+impl Dataset {
```
Comment (kiavash-at-work) [inline shavee-core/src/zfs.rs:9]: Functions now are changed to methods.

**Comment by kiavash-at-work** [comment]
These are remaining tasks marked in the code as: TODO

* Wait for `clap_config` crate to be ported to clap 3.0 so config file can be implemented (Issue #16)
* Need to implement integration test for `main.rs`
* Do the simple functions in `logic.rs` need unit tests?
* How to implement unit test for Yubikey which needs human interaction?
* `In zfs.rs` Modify the method function to return **Self** to enable chaining actions like `dataset.loadkey(passphrase).mount()`
* All functions in PAM module needs unit tests.

**Comment by kiavash-at-work** [comment]
This PR is ready for merge. I will open new trackers for remaining TODO lists

**Comment by ashuio** [comment]
Sorry for the delay with the merge, Thanks for your contribution

**Pull request merged**